src/script/descriptor.cpp

```diff
@@ -1524,11 +1541,12 @@ class StringMaker {
     StringMaker(const SigningProvider* arg LIFETIMEBOUND, const std::vector<std::unique_ptr<PubkeyProvider>>& pubkeys LIFETIMEBOUND, bool priv)
         : m_arg(arg), m_pubkeys(pubkeys), m_private(priv) {}
 
-    std::optional<std::string> ToString(uint32_t key) const
+    std::string ToString(uint32_t key, bool* has_priv_key = nullptr) const
     {
         std::string ret;
         if (m_private) {
-            if (!m_pubkeys[key]->ToPrivateString(*m_arg, ret)) return {};
+            assert(has_priv_key != nullptr);
```

Comment:
In https://github.com/bitcoin/bitcoin/commit/deed439bff638abfc298a3b9c4f3d9f75fa18244 "descriptor: ToPrivateString() pass if at least 1 priv key exists"

Looks like the same argument applies here too?

https://github.com/bitcoin/bitcoin/pull/32471#discussion_r2258451488

> If has_priv_key is never supposed to be null, then it should be a reference.